spec-tests/src/test/java/org/candlepin/spec/pools/OneSubPoolPerStackSpecTest.java

```diff
@@ -0,0 +1,805 @@
+    }
+
+    private void createTargetPoolAndProducts(OwnerDTO owner, String stackingId) {
+        ContentDTO ownerContent1 = ownerContentApi.createContent(owner.getKey(), Content.random());
+        ContentDTO ownerContent2 = ownerContentApi.createContent(owner.getKey(), Content.random());
+        ProductDTO ownerDerivedEngProduct = ownerProductApi.createProductByOwner(
+            owner.getKey(), Products.random());
+        ownerProductApi.addBatchContent(owner.getKey(), ownerDerivedEngProduct.getId(),
+            Map.of(ownerContent1.getId(), true, ownerContent2.getId(), true));
+
+        ProductDTO ownerDerivedSku = ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.random()
+            .multiplier(1L)
+            .attributes(List.of(new AttributeDTO().name("host_limited").value("true"))));
+
+        ProductDTO ownerSku = ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.random()
+            .multiplier(1L)
+            .attributes(List.of(
+                new AttributeDTO().name("host_limited").value("true"),
+                new AttributeDTO().name("stacking_id").value(stackingId),
+                new AttributeDTO().name("virt_limit").value("unlimited"))));
+
+        ownerApi.createPool(owner.getKey(), Pools.random(ownerSku)
+            .stackId(stackingId)
+            .stacked(true)
+            .startDate(OffsetDateTime.now().minusSeconds(5L))
+            .derivedProductId(ownerDerivedSku.getId())
+            .derivedProvidedProducts(Set.of(
+                new ProvidedProductDTO()
+            .productId(ownerDerivedEngProduct.getId())
+            .productName(ownerDerivedEngProduct.getName()))));
+    }
+
+    private String getAttributeValue(String name, List<AttributeDTO> attributes) {
+        List<AttributeDTO> matches = attributes.stream()
+            .filter(x -> name.equals(x.getName()))
+            .collect(Collectors.toList());
+        if (matches.size() > 0) {
+            return matches.get(0).getValue();
+        }
+        return null;
+    }
+
+    private List<PoolDTO> getHostPools(TestData data, List<String> contracts) {
+        return data.getHostClient().pools().listPoolsByConsumer(data.getHost().getUuid()).stream()
+            .filter(item -> contracts.contains(item.getContractNumber()))
+            .sorted(Comparator.comparing(item -> contracts.indexOf(item.getContractNumber())))
+            .collect(Collectors.toList());
+    }
+
+    private EntitlementDTO bindHostPool(TestData data, List<PoolDTO> pools, String contract) {
+        PoolDTO pool = pools.stream()
+            .filter(x -> contract.equals(x.getContractNumber()))
+            .collect(Collectors.toList()).get(0);
+        return ApiClient.MAPPER.convertValue(
+            data.getHostClient().consumers().bindPool(data.getHost().getUuid(), pool.getId(), 1).get(0),
+            EntitlementDTO.class);
+    }
+
+    private PoolDTO getSubPool(TestData data) {
```

Comment:
Instead of using TestData as a parameter here, you could use the consume uuid String and make the method more versatile.

Reply:
I use more than just the uuid.

Reply:
That is my bad, I didn't see that you were also using the stackId1 as well